planner/core/rule_build_key_info.go

```diff
@@ -91,6 +94,23 @@ func (p *LogicalSelection) checkMaxOneRowCond(unique expression.Expression, cons
 	return okCorCol
 }
 
+func isUnique(p *LogicalSelection, col *expression.Column) bool {
```

Comment:
Add some comments to tell others that the `equal` function rejects null values. So we can also check the indexes who is unique key without not null restriction.

Reply:
OK，done.